Bugfix regarding unaliased services

boesing opened the pull request:
|    Q          |   A
|-------------- | ------
| Bugfix        | yes

### Description

This is a fix for the related issue #51 

This change will add several, missing, aliases to the configuration after the post processor was executed. This will provide aliases for non-migrated packages which might be still required in an already migrated package.

boesing committed:
Added failing unit test

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

boesing committed:
Provided bugfix for services which does not have an alias

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

boesing committed:
Removed unused code and optimized code execution

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

boesing committed:
Re-adding replaced aliases and invokables before passing to the factory replacement

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

boesing committed:
Added some more test stuff to increase code coverage

Signed-off-by: Maximilian Bösing <2189546+boesing@users.noreply.github.com>

weierophinney reviewed (approved):
:ship:

michalbundyra commented on src/ConfigPostProcessor.php:
```diff
@@ -260,4 +278,35 @@ private function noopReplacement($value)
+    private function replaceDependencyFactories(array $config)
+    {
+        $factories = isset($config['factories']) ? $config['factories'] : [];
+        $aliases = isset($config['aliases']) ? $config['aliases'] : [];
```
we don't need these variables, in the loop we can just do:

```php
if (!isset($config['aliases'][$service'])) {
    $config['aliases'][$service'] = $replacedService;
}
```

and then we can also remove after the loop:

```php
       if ($aliases !== []) {
             $config['aliases'] = $aliases;
         }
```

the same with `$factories`.

It should reduce complexity and additional variables here.

weierophinney committed:
refactor: internals of ConfigPostProcessor::replaceDependencyFactories

Works directly on the `$config` members, instead of extracting them, and
then resetting them.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
fix: ensure factory name is also rewritten

Previously, we were rewriting the factory name as well if we could, but
hadn't captured this fact in tests. This patch adds a test, and changes
the behavior of the primary patch for #52 to implement the behavior.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
docs: Adds CHANGELOG entries for #52

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: change test case to reflect real-world scenario

The actual scenario also involved a class name `ZendFormFactory`. The
class name should not be rewritten.

(This test case currently fails, but it's unrelated to #52, so we will
merge it, and introduce another patch to fix it.)

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

michalbundyra commented:
Thanks, @boesing!
